airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/JdbcBufferedConsumerFactory.kt

```diff
@@ -145,11 +145,14 @@ object JdbcBufferedConsumerFactory {
     private fun onStartFunction(
```

Comment:
what do you think of having this function return a `Pair<OnStartFunction, List<WriteConfig>>`?
```kotlin
fun onStartFunction(): Pair<OnStartFunction, List<WriteConfig>> {
  val writeConfigs = mutableList()
  val onStartFunction = OnStartFunction { ... populate writeConfigs.... }
  return Pair(onStartFunction, writeConfigs)
}
```

and then in `createAsync`, we don't need to create a mutable list manually:
```kotlin
val (onStartFunction, writeConfigs) = onStartFunction(...)
return AsyncStreamConsumer(
  outputRecordCollector,
  onStartFunction,
  JdbcFlushFunction(...writeConfigs...)
  ...
)
```